### PR TITLE
Implement (*SessionProtocol).Close()

### PR DESCRIPTION
**Pull request by 0x1a8510f2**
I noticed that a HTTP server listening on pinecone cannot be gracefully shut down or restarted because pinecone does not implement the `Close()` method of `net.Listener`.

This PR also includes a minor tweak and deps updates detailed below:

- d31752ec1cf7630fddb91a6eb974b599e75b8794: Edit a log line left over from before the utp -> QUIC switch.
- fc6bc1056ecacfe16214491e8dea8dcb2ddce602: Actually implement the `Close()` method.
- c5f994c9360a2922576223e9924ea32547f58aed: It seems as though calling `Shutdown()` on a HTTP server actually closes the listener twice ([here](https://cs.opensource.google/go/go/+/refs/tags/go1.18.4:src/net/http/server.go;l=2750) and [here](https://cs.opensource.google/go/go/+/refs/tags/go1.18.4:src/net/http/server.go;l=3016)) so this needs to be handled. Not sure if this is working as intended or an stdlib bug.
- a38ec944325fd171da9fa0c1d525b7af84a2301f: Update dependencies and run `go mod tidy`. I did this because trying to use a newer version of `quic-go` in my project than that used by pinecone causes segfaults. Feel free to leave the commit out. Just didn't think it warranted a separate PR. Didn't seem to cause any issues in my testing.

Signed-off-by: `0x1a8510f2 <admin@0x1a8510f2.space>`

### CODE REVIEW

**Review by neilalexander — APPROVED** [review]

**Comment by 0x1a8510f2** [comment]
Strange. I'm sure I tested this. Anyway, I'm currently on holiday but will take a look soon.

**Comment by 0x1a8510f2** [comment]
Found the problem: https://github.com/lucas-clemente/quic-go/commit/e71c236232cad376b84f330cb521350f5ce8f4f3

Fixed in b90247a3a074b709a3bde33c1a941471c372290c :D

File: sessions/sessions.go [hunk 1]
```diff
@@ -51,6 +51,8 @@ type SessionProtocol struct {
 	proto    string
 	streams  chan net.Conn
 	sessions sync.Map // types.PublicKey -> *activeSession
+	mutex    sync.Mutex
+	closed   bool
```
Comment (neilalexander) [inline sessions/sessions.go:55]: It might be easier/cleaner to use a `sync.Once` than a `sync.Mutex`+`bool`: https://pkg.go.dev/sync#Once

**Review by neilalexander — APPROVED** [review]
Overall seems fine.

File: sessions/sessions.go (hunk 1 above)
Comment (0x1a8510f2) [reply sessions/sessions.go]: True. My reasoning was that this provides an easy way to check if the session is closed but as that isn't really being used, I've changed it.

**Pull request merged**